scripts/spack/packages/axom/package.py

```diff
@@ -17,6 +17,52 @@
 )
 from spack_repo.builtin.build_systems.cuda import CudaPackage
 from spack_repo.builtin.build_systems.rocm import ROCmPackage
+from spack.error import SpackError
+
+# Axom components we expose to Spack.  Core is always built and is not listed here.
+_AXOM_COMPONENTS = (
+    "bump",
+    "inlet",
+    "klee",
+    "lumberjack",
+    "mint",
+    "mir",
+    "multimat",
+    "primal",
+    "quest",
+    "sidre",
+    "sina",
+    "slam",
+    "slic",
+    "spin",
+)
+
+# Hard inter-component dependencies taken from Axom's dependency graph.
+# Only *solid* edges from dependencies.dot are encoded here
+_AXOM_COMPONENT_DEPS = {
+    "bump": {"slic", "spin", "primal"},
+    "inlet": {"sidre", "slic", "primal"},
+    "klee": {"sidre", "slic", "inlet", "primal"},
+    # lumberjack only depends on core
+    "mint": {"slic", "slam"},
+    "mir": {"bump", "slic", "slam", "primal"},
+    "multimat": {"slic", "slam"},
+    "primal": {"slic"},
+    "quest": {"slic", "slam", "primal", "mint", "spin"},
+    "sidre": {"slic"},
+    "sina": {"slic"}, # sina's tests depends on slic
+    "slam": {"slic"},
+    # slic only depends on core
+    "spin": {"slic", "slam", "primal"},
+}
```

Comment:
I foresee these being broken out as dependencies one day, but if you're going to do it this way with a variant, why not do what Trilinos has done before -- implement these dependencies as requirements so that you will not see them combined in ways you don't expect, e.g.:

```python
requires("components=slic,spin,primal", when="components=bump")
requires("components=sidre,slic,primal", when="components=inlet")
...
```

Then you don't have to validate them. The concretizer will validate them.

Reply:
Thanks @tgamblin -- I suspected there was a better way to do this!